sqlserver/datadog_checks/sqlserver/statements.py

```diff
@@ -390,6 +416,12 @@ def _collect_plans(self, rows, cursor, deadline):
                 self.check.count("dd.sqlserver.statements.deadline_exceeded", 1, **self.check.debug_stats_kwargs())
                 return
             plan_key = (row['query_signature'], row['query_hash'], row['query_plan_hash'])
+            # for stored procedures, we only want to look up plans for the entire procedure
+            # not every query that is executed within the proc. Queries from the same procedure
+            # have the same plan_handle, but different query signatures & can have diff plan hashes.
+            # taking the first 16 bytes should be unique enough to properly identify the plan.
+            if row['is_proc']:
+                plan_key = row['plan_handle'][0:16]
```

Comment:
In my tests, I found that the `query_plan_hash` for queries executed in the same procedure were sometimes the same, and sometimes different. It wasn't consistent. The only thing consistent for all of them was the `plan_handle`, so I am using the first 16 chars as the key.